computation/src/main/java/com/powsybl/computation/CompletableFutureTask.java

```diff
@@ -0,0 +1,90 @@
+
+/**
+ *
+ * A {@link CompletableFuture} which embeds an actual task, unlike the default implementation.
+ * In particular, the behaviour of {@link #cancel(boolean)} is modified in order to actually
+ * interrupt the thread executing the bound task.
+ *
+ * <p>Similarly to {@link FutureTask}, this class implements {@link Runnable} and
+ * can therefore be submitted to an {@link java.util.concurrent.Executor} for execution.
+ *
+ * @author Sylvain Leclerc <sylvain.leclerc at rte-france.com>
+ */
+public class CompletableFutureTask<R> extends CompletableFuture<R> implements Runnable {
+
+    private final FutureTask<R> future;
+
+    /**
+     * Creates a completable future bound to the specified task. This task will be executed
+     * on a call to {@link #run()}, for instance by submitting this object to
+     * an {@link java.util.concurrent.Executor}.
+     *
+     * @param task The task to be bound to this future.
+     */
+    public CompletableFutureTask(Callable<R> task) {
+        future = new FutureTask<>(task);
+    }
+
+    /**
+     * Executes the wrapped callable. On execution end, this future will complete with the callable result,
+     * or complete exceptionally in case of exception.
+     */
+    @Override
+    public void run() {
+        future.run();
+        try {
+            complete(future.get());
+        } catch (ExecutionException exc) {
+            completeExceptionally(exc.getCause());
+        } catch (Throwable exc) {
+            completeExceptionally(exc);
+        }
+    }
+
+    /**
+     * Submits this task to the specified executor.
+     * @param executor The executor which will execute this task.
+     * @return         This.
+     */
+    public CompletableFutureTask<R> runAsync(Executor executor) {
+        executor.execute(this);
+        return this;
+    }
+
+    /**
+     * Creates a completable future task and submits it to the specified executor.
+     * @param task     The task to be executed.
+     * @param executor The executor which will execute the task.
+     * @return         A {@link CompletableFutureTask} bound to the specified task.
+     */
+    public static <T> CompletableFutureTask<T> runAsync(Callable<T> task, Executor executor) {
+        return new CompletableFutureTask<>(task).runAsync(executor);
+    }
+
+    /**
+     * Cancels this completable future and, if requested and if the bound task is under execution,
+     * interrupts the thread executing the task.
+     *
+     * @param mayInterruptIfRunning if {@code true}, and if the bound task is under execution,
+     *                              the thread executing the task will be interrupted
+     * @return {@code true} if the task has been cancelled.
+     */
+    @Override
+    public boolean cancel(boolean mayInterruptIfRunning) {
+        boolean cancelled = super.cancel(mayInterruptIfRunning);
+        if (cancelled) {
+            future.cancel(mayInterruptIfRunning);
+        }
+        return cancelled;
```

Comment:
Was it on purpose that you return the super::cancel instead of the futuretask::cancel result ? Which one should we return ? and should we insist on the difference ?

I like the behavior of futureTask better, but we are returning a completablefuture so behaving like one is maybe better in this case ?

Reply:
I must say it was not on purpose.

This implementation of the method anyway already does not respect the javadoc of `CompletableFuture` since it actually interrupts an underlying thread. Maybe we could keep the whole package of the `Future` behaviour ... I propose to change it in this direction, although it's not perfect. Anyway the javadoc and the implementation are not consistent here.